packages/mobile/src/identity/securityCode.ts

```diff
@@ -1,21 +1,26 @@
 import {
   ActionableAttestation,
-  AttestationServiceSecurityCodeRequest,
   AttestationsWrapper,
 } from '@celo/contractkit/lib/wrappers/Attestations'
 import { PhoneNumberHashDetails } from '@celo/contractkit/src/identity/odis/phone-number-identifier'
+import { GetAttestationRequest } from '@celo/utils/lib/io'
 import { Address } from '@celo/utils/src/address'
 import BigNumber from 'bignumber.js'
 import Logger from 'src/utils/Logger'
 
 const TAG = 'identity/securityCode'
 
+const cache: { [key: string]: string } = {}
+
 function hashAddressToSingleDigit(address: Address): number {
   return new BigNumber(address.toLowerCase()).modulo(10).toNumber()
 }
 
 export function getSecurityCodePrefix(issuerAddress: Address) {
-  return `${hashAddressToSingleDigit(issuerAddress)}`
+  if (!cache[issuerAddress]) {
+    cache[issuerAddress] = `${hashAddressToSingleDigit(issuerAddress)}`
+  }
+  return cache[issuerAddress]
```

Comment:
I don't actually think this is providing that much gain, hashAddressToSingleDigit does seem to computationally complex. I was thinking we could have a cache like you originally suggested that was `key:digit --> List<IssuerAddresses>`. That way when you receive a code, you can just get the check the cache for the list of issuers to send to rather than iterating through each issuer. Though, it would require more maintenance -- eg. removing entries whenever you remove entries from the existing ActiveAttestations cache

Reply:
I would not include it in the current PR. We can handle it later as an improvement to the current solution.